### ISSUE TITLE
Caching Pipes

### ISSUE THREAD

**Issue by WASasquatch** [huggingface/diffusers#1253]
### Describe the bug

It used to possible to cache stable Diffusion pipes to load them faster, and more efficiently. But development in diffusers has got in the way of this working smoothly. 

> If a model is already downloaded, it still checks HF model conf, and this process is needlessly slow
> It Is now not possible to load a cached pipe from pickle/joblib when you restart a Colab session or start a new one. In the past, I was able to load the same pipe for a month+ through various diffusers updates. The reason this happens is it cannot import "diffusers_modules" which is not a real importable file and is referenced dynamically in diffusers.utils.__init__.py

It would be awesome If this method of multi-pipe handling with pickle or joblib could still be viable, and a fast option. Initially, before checking HF model conf, a pipe could load in 2 seconds. Now it takes anywhere from 15-25s which compared to the requested diffusion time of 1s is really slow. It was a very fast method to handling multiple models and pipes. 

### Reproduction

Dump a diffusion pipeline with joblib or pickle and load it in a new session or env. 

### Logs

```shell
N/A
```

### System Info

Will Update (on phone) but it's usually colab environment with T4

**Comment by patrickvonplaten** [comment]
Hey @WASasquatch,

Could you please copy-paste a code snippet we can run and compare to? Loading a cached pipeline should not take 30 seconds. What might help is to pass `local_files_only=True` so that it never checks whether the config has been updates only (see here: https://huggingface.co/docs/diffusers/api/diffusion_pipeline#diffusers.DiffusionPipeline.from_pretrained.local_files_only(bool, )

**Comment by WASasquatch** [comment]
Yeah I do know about `local_files_only` but this is sorta unintuitive as it's basically really just `offline_mode=True` as you then can't download ***any*** models. So you have to manually disable to download a new model.

And to test simple

Install joblib
`pip install joblib`

And then dump a pipe and load it in a new python session (windows), or migrate the pipe to another system (same error, probably more issues though env wise)
```python
joblib.dump(StableDiffusionPipeline.from_pretrained( params... ), './sdv1-5_full-float.pipe')

# ...

pipe = joblib.load('./sdv1-5_full-float.pipe')
```
